backend/app/Actions/User/RegisterUserAction.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\Actions\User;
+
+use app\Repositories\Contracts\UserRepository;
+use App\User;
+use Tymon\JWTAuth\Facades\JWTAuth;
+
+class RegisterUserAction
+{
+    private $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function execute(RegisterRequest $request): RegisterResponse
+    {
+        $user = $this->userRepository->save(
+            User::create([
```

Comment:
if you call "create" the user will be stored to DB instantly. So you should create a new model ($user = new User;), then assignt properties $user->name = $request->getName(); and so on, and finally call save method of repository $userRepository->save($user);. Such actions are needed because of all actions you do with persisting data should be inside repositories